ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -163,10 +164,23 @@ public WSQueryEndpoint(
         Objects.requireNonNull(serviceContextFactory, "serviceContextFactory");
   }
 
+  private void checkEndpointAuthorization(final Principal userPrincipal) {
+    final Class className = this.getClass();
+    final String methodName = this.getClass().getAnnotation(ServerEndpoint.class).value();
+
+    if (!securityExtension.getAuthorizer().hasAccess(userPrincipal, className, methodName)) {
+      final String userName = (userPrincipal != null) ? userPrincipal.getName() : null;
+      throw new KsqlException(
+          String.format("User:%s is denied to access this cluster.", userName)
+      );
```

Comment:
Besides throwing an error we want to close the session with the appropriate error code, right? I see that the unit test you've added checks for the close code `CANNOT_ACCEPT` which makes sense to me. Where is that close code being returned?

Reply:
The exception is caught un the `inOpen` method. The try/catch that catches it makes sure to close the session:

```
try {
      // Check if the user has authorization to access this Websocket endpoint
      checkEndpointAuthorization(session.getUserPrincipal());
      ....
} catch (final Exception e) {
      log.debug("Error processing request", e);
      SessionUtil.closeSilently(session, CloseCodes.CANNOT_ACCEPT, e.getMessage());
}
```